src/js/xAxisTimeSeries.js

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    // Mix-in grade for time-series x axis
+
+    fluid.defaults("floe.chartAuthoring.xAxisTimeSeries", {
+        model: {
+            dataSet: []
+        },
+        axisOptions: {
+            numberOfXAxisTicks: 6
+        },
+        selectors: {
+            xAxis: ".floec-ca-lineChart-x-axis"
+        },
+        events: {
+            onChartCreated: null,  // Fire when the line is created. Ready to register D3 DOM event listeners,
+            onDraw: null
+        },
+        listeners: {
+            "onDraw.drawXAxis": {
+                func: "{that}.drawXAxis",
+                priority: "before:drawChartLine"
+            }
+        },
+        invokers: {
+            drawXAxis: {
+                funcName: "floe.chartAuthoring.xAxisTimeSeries.drawXAxis",
+                args: ["{that}"]
+            },
+            getXAxisTickFormat: {
+                funcName: "floe.chartAuthoring.xAxisTimeSeries.getXAxisTickFormat"
+            }
+        }
+    });
+
+    floe.chartAuthoring.xAxisTimeSeries.drawXAxis = function (that) {
+        var xAxisClass = that.classes.xAxis,
+            padding = that.options.lineOptions.padding,
+            height = that.options.svgOptions.height,
+            axisTransform = "translate(0," + (height - padding) + ")",
+            xAxis = floe.chartAuthoring.xAxisTimeSeries.getXAxis(that);
+
+        that.manageAxis("xAxis", xAxisClass, axisTransform, xAxis);
+    };
+
+    floe.chartAuthoring.xAxisTimeSeries.getXAxis = function (that) {
```

Comment:
This method seems generic as to whether the X axis represents a time series or not